modules/appeals_api/app/models/appeals_api/non_veteran_claimant.rb

```diff
@@ -48,21 +48,33 @@ def number_and_street
     end
 
     def city
-      form_data['address']['city']
+      form_data.dig('address', 'city')
     end
 
     def state_code
-      form_data['address']['stateCode']
+      form_data.dig('address', 'stateCode')
     end
 
     def country_code
+      return if form_data.blank?
+
       form_data['address']['countryCodeISO2'] || 'US'
     end
 
     def zip_code_5
+      return if form_data.blank?
+
       form_data['address']['zipCode5'] || '00000'
     end
 
+    def zip_code
+      if zip_code_5 == '00000'
```

Comment:
I think this should work instead of this kind of convoluted `if` structure:
```ruby
form_data.dig('address', 'internationalPostalCode') || zip_code_5
```

Reply:
I don't disagree that it's ugly, but we want to replace the internationalPostalCode with '00000' if it's been supplied for zip_code5, but print the international postal code if it is present, I believe.

Reply:
Now I'm replaying this thing in my head...

Reply:
So if there is no international postal code, and zip_code_5 is '48323', then '48323' will print for zip code and international postal code, right?

Reply:
If the zip_code is `00000` then we want to print the `internationalPostalCode` on the pdf - but we have to send `00000` to CMP, unless I'm misremembering something.

Reply:
We don't differentiate on the form between zipcode & international postal code - the same field is in use for both.

So, if they've supplied an international postal code we will use it, otherwise we'll use the provided zip_code_5, which itself includes a fallback to print '00000' if it is missing. I think the logic still works out for my shortened version.

Reply:
@nathanbwright Correct, but we [use zip_code_5](https://github.com/department-of-veterans-affairs/vets-api/blob/74294c7c0590905e4d9254e2c03acc66274c62e2/modules/appeals_api/lib/appeals_api/hlr_pdf_submit_wrapper.rb#L13) when talking to centralmail to default to 00000

Reply:
> It's looking good! It looks the claimant first and last name aren't totally wired up for printing on the pdf... Just the middle name is printed on the `expected_200996_v2_extra.pdf` file.

The stamped and auto-sized fields are coming in the next PR!

Reply:
I'm still confused on this one. I thought the zeros were only to indicate outside of US.

```
"zipCode5": {
      "type": "string",
      "description": "5-digit zipcode. Use \"00000\" if Veteran is outside the United States",
      "pattern": "^[0-9]{5}$"
},
 "internationalPostalCode": { "type": "string", "maxLength": 16 }
 ```

Reply:
ok. i've confirmed that we do in fact assign `00000` if there is no zip code. so do we prefer this? printing potentially nothing under claimant and then randomly a '00000' for zip code?

Reply:
Final report! We assign the zeros for a  _Veterant if_ they are experiencing homelessness. This will not work the same way for a claimant. If they provide any address info, the zipCode5 field is required, and it will come in as a domestic zip, or '00000', indicating an internationalPostalCode. So, it should be as simple as, zipCode5 || InternationalPostalCode.

Reply:
I think asking for 0's like we do for veterans is fine :)

Reply:
but why ask claimants for zeros? why not supply one or the other?